rust_bitcoin_lib/src/lib.rs

```diff
@@ -100,6 +101,23 @@ pub unsafe extern "C" fn rust_bitcoin_psbt(data: *const u8, len: usize) -> *mut
     }
 }
 
+#[no_mangle]
+pub unsafe extern "C" fn rust_bitcoin_script(data: *const u8, len: usize) -> bool {
+    // Safety: Ensure that the data pointer is valid for the given length
+    let data_slice = slice::from_raw_parts(data, len);
+
+    let script: Result<(bitcoin::script::ScriptBuf, usize), encode::Error> = encode::deserialize_partial(data_slice);
+    match script {
+        Err(_) => false,
+        Ok(s) => {
+           if s.0.is_op_return() || s.0.len() > 10_000 {
```

Comment:
I think there are unintentional *minor* formatting issues in this line

Reply:
which one?

Reply:
got it, fixed